### ISSUE TITLE
HeaderNotification - close button's hover state not distinguishable from surrounding div

### ISSUE THREAD

**Issue by jendowns** [carbon-design-system/ibm-security#435]
**Expected behavior -**

When you hover over the "close" button for an individual notification in a header popover, that hover state should be distinguishable from the hover state of the surrounding div.

**Actual behavior -**

When you hover over a notification (i.e., the whole notification "row" or wrapping div), then the notification row's background color changes and the close button appears.

However, when you then move your mouse over the close button for that notification, the button itself should have its own hover state that is distinguishable from the hover state of the surrounding div. The button's hover state has no additional change, and the cursor is also a default/auto cursor, not a pointer.

In other words: there's no visual indication to users that they can click on the close button, via a background color + cursor change.

### Steps for reproducing

https://ibm-security.carbondesignsystem.com/?path=/story/patterns-header--with-notifications
